presto-main/src/main/java/com/facebook/presto/server/ServerInfoResource.java

```diff
@@ -38,12 +38,14 @@
 public class ServerInfoResource
 {
     private final ServerInfo serverInfo;
+    private final ServerConfig serverConfig;
     private final GracefulShutdownHandler shutdownHandler;
 
     @Inject
-    public ServerInfoResource(ServerInfo serverInfo, GracefulShutdownHandler shutdownHandler)
+    public ServerInfoResource(ServerInfo serverInfo, ServerConfig serverConfig, GracefulShutdownHandler shutdownHandler)
     {
         this.serverInfo = requireNonNull(serverInfo, "serverInfo is null");
+        this.serverConfig = requireNonNull(serverConfig, "serverConfig is null");
```

Comment:
Configs are mutable and should not be stored. Instead, extract the needed information in the constructor